python/ray/data/datasource/avro_datasource.py

```diff
@@ -0,0 +1,53 @@
+from io import BytesIO
+from typing import TYPE_CHECKING, Iterator, List, Union
+
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
+from ray.data.block import Block
+from ray.data.datasource.file_based_datasource import FileBasedDatasource
+from ray.util.annotations import PublicAPI
+
+if TYPE_CHECKING:
+    import pyarrow
+
+
+@PublicAPI(stability="alpha")
+class AvroDatasource(FileBasedDatasource):
+    """Avro datasource, for reading and writing Avro files."""
+
+    _FILE_EXTENSIONS = ["avro"]
+
+    def __init__(
+        self,
+        paths: Union[str, List[str]],
+        **file_based_datasource_kwargs,
+    ):
+        super().__init__(paths, **file_based_datasource_kwargs)
+
```

Comment:
(Also add `from ray.data._internal.util import _check_import`)
```suggestion

        _check_import(self, module="fastavro", package="fastavro")
```

Reply:
added